cassiopeia/solver/GreedySolver.py

```diff
@@ -136,7 +136,7 @@ def _solve(
             )
 
         # extract character matrix
-        character_matrix = cassiopeia_tree.get_original_character_matrix()
+        character_matrix = cassiopeia_tree.get_current_character_matrix()
```

Comment:
I think we probably do want to use the current character matrix, as the user could for example infer missing data on the matrix beforehand and we would want to solve the post-processed CM

Reply:
That's a fair point. The question is if the user would already have a tree in the `CassiopeiaTree` object when we run it through a solver. If you take a look at the `set_character_matrix` code, we overwrite both character matrices. The only time the current and original character matrix would diverge is when states are set on particular leaves, and this can only happen if we have a tree already in the object.

All in all, this a good example of when having two types of character matrices is actually quite confusing!